ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -0,0 +1,211 @@
+        metrics.daml.services.index.streamsBuffer.transactionTreeEventsResolvedFromBuffer,
+      totalRetrievedCounter =
+        metrics.daml.services.index.streamsBuffer.totalTransactionTreesRetrieved,
+      bufferSizeCounter =
+        // TODO in-memory fan-out: Specialize the metric per consumer
+        metrics.daml.services.index.transactionTreesBufferSize,
+      outputStreamBufferSize = outputStreamBufferSize,
+    )
+
+  override def lookupFlatTransactionById(
+      transactionId: TransactionId,
+      requestingParties: Set[Party],
+  )(implicit loggingContext: LoggingContext): Future[Option[GetFlatTransactionResponse]] =
+    delegate.lookupFlatTransactionById(transactionId, requestingParties)
+
+  override def lookupTransactionTreeById(
+      transactionId: TransactionId,
+      requestingParties: Set[Party],
+  )(implicit loggingContext: LoggingContext): Future[Option[GetTransactionResponse]] =
+    delegate.lookupTransactionTreeById(transactionId, requestingParties)
+
+  override def getActiveContracts(activeAt: Offset, filter: FilterRelation, verbose: Boolean)(
+      implicit loggingContext: LoggingContext
+  ): Source[GetActiveContractsResponse, NotUsed] =
+    delegate.getActiveContracts(activeAt, filter, verbose)
+
+  override def getContractStateEvents(startExclusive: (Offset, Long), endInclusive: (Offset, Long))(
+      implicit loggingContext: LoggingContext
+  ): Source[((Offset, Long), ContractStateEvent), NotUsed] =
+    throw new UnsupportedOperationException(
+      s"getContractStateEvents is not supported on ${getClass.getSimpleName}"
+    )
+
+  override def getTransactionLogUpdates(
+      startExclusive: (Offset, EventSequentialId),
+      endInclusive: (Offset, EventSequentialId),
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[((Offset, EventSequentialId), TransactionLogUpdate), NotUsed] =
+    throw new UnsupportedOperationException(
+      s"getTransactionLogUpdates is not supported on ${getClass.getSimpleName}"
+    )
+}
+
+private[platform] object BufferedTransactionsReader {
+  type FetchTransactions[FILTER, API_RESPONSE] =
+    (Offset, Offset, FILTER, Boolean) => Source[(Offset, API_RESPONSE), NotUsed]
+
+  def apply(
+      delegate: LedgerDaoTransactionsReader,
+      transactionsBuffer: EventsBuffer[Offset, TransactionLogUpdate],
+      lfValueTranslation: LfValueTranslation,
+      metrics: Metrics,
+  )(implicit
+      loggingContext: LoggingContext,
+      executionContext: ExecutionContext,
+  ): BufferedTransactionsReader =
+    new BufferedTransactionsReader(
+      delegate = delegate,
+      transactionsBuffer = transactionsBuffer,
+      toFlatTransaction =
+        TransactionLogUpdatesConversions.ToFlatTransaction(_, _, _, lfValueTranslation),
+      toTransactionTree =
+        TransactionLogUpdatesConversions.ToTransactionTree(_, _, _, lfValueTranslation),
+      metrics = metrics,
+    )
+
+  private[events] def getTransactions[FILTER, API_TX, API_RESPONSE](
```

Comment:
I am not sure what the benefit of this parameter currying is when we are not using this function as partially applied.

Reply:
I can't argue with more than "cohesion of the argument list groups". If you think it makes the code more readable with one arg list, I can change it

Reply:
That's fine